test/framework/control_plane.go

```diff
@@ -69,81 +78,118 @@ func (input *ControlplaneClusterInput) ControlPlaneCluster() {
 	By("creating an InfrastructureCluster resource")
 	Expect(mgmtClient.Create(ctx, input.InfraCluster)).To(Succeed())
 
+	// This call happens in an eventually because of a race condition with the
+	// webhook server. If the latter isn't fully online then this call will
+	// fail.
 	By("creating a Cluster resource linked to the InfrastructureCluster resource")
 	Eventually(func() error {
-		err := mgmtClient.Create(ctx, input.Cluster)
-		if err != nil {
-			fmt.Println(err)
+		if err := mgmtClient.Create(ctx, input.Cluster); err != nil {
+			fmt.Printf("%+v\n", err)
+			return err
 		}
-		return err
-	}, input.CreateTimeout, 10*time.Second).Should(BeNil())
+		return nil
+	}, input.CreateTimeout, eventuallyInterval).Should(BeNil())
+
+	By("creating related resources")
+	for _, obj := range input.RelatedResources {
+		By(fmt.Sprintf("creating a/an %s resource", obj.GetObjectKind().GroupVersionKind()))
+		Eventually(func() error {
```

Comment:
Is this to cover the case when we store related resources that are using the capi webhook servers?

Reply:
No, this is the load balancer.

Reply:
But since it’s a generic resource, I figure an eventually makes sense.